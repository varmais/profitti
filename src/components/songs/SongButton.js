import React, { Component } from 'react';
import NavigationButton from '../navigation/NavigationButton';
import { navigatorPropTypes, songPropTypes } from '../../helpers/PropTypes';

export default class SongButton extends Component {
  static propTypes = {
    song: songPropTypes(),
    navigation: navigatorPropTypes()
  };

  render () {
    const { song } = this.props;
    return <NavigationButton text={`${song.id} ${song.title}`} onPress={this.handlePress} />;
  }

  handlePress = () => {
    const { song, navigation } = this.props;
    navigation.navigate('Song', {song});
  };
}