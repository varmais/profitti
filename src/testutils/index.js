import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}