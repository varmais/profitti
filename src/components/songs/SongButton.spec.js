import { createNavigationProp } from '../../testutils';
import SongButton from './SongButton';

describe('components:songs:SongButton', () => {
  const song = {
    id: 123,
    title: 'song title'
  };

  test('renders SongButton', () => {
    const snapshot = renderer.create(<SongButton
      song={song}
      navigation={createNavigationProp()}
    />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('when button is pressed', () => {
    test('it navigates to song screen');
  });
});