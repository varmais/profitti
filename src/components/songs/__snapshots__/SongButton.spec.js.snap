// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:songs:SongButton renders SongButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 18,
          },
        ]
      }
    >
      123 song title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "bottom": -2,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      Ôèë
    </Text>
  </View>
</View>
`;
